deepautoencoder/stacked_autoencoder.py

```diff
@@ -127,10 +140,47 @@ def run(self, data_x, data_x_, hidden_dim, activation, loss, lr,
                 print('epoch {0}: global loss = {1}'.format(i, l))
         # debug
         # print('Decoded', sess.run(decoded, feed_dict={x: self.data_x_})[0])
-        self.weights.append(sess.run(encode['weights']))
-        self.biases.append(sess.run(encode['biases']))
+
```

Comment:
This breaks the code inside transform(). Because you are appending tf.Variable object to self.weights but transform() assumes that the weights are in a numpy array. You may want to somehow modify transform() to operate on tf.Variable by running a session.